torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -107,6 +107,31 @@ class SubgraphSlicer {
       }
     }
 
+    // merge adjacent subgraphs thaat don't have a dependency on each other
+    Node* prev_autodiff_group = nullptr;
```

Comment:
maybe you could a comment here specifying that `prev` here means previous in terms of the iteration order, but it's actually next in terms of IR order.